presto-orc/src/main/java/com/facebook/presto/orc/reader/DoubleSelectiveStreamReader.java

```diff
@@ -56,29 +56,29 @@ public class DoubleSelectiveStreamReader
 
     private final StreamDescriptor streamDescriptor;
     @Nullable
-    private final TupleDomainFilter filter;
-    private final boolean nonDeterministicFilter;
-    private final boolean nullsAllowed;
-    private final boolean outputRequired;
+    protected final TupleDomainFilter filter;
```

Comment:
I don't love the idea of exposing all of these internal details to the whole package. I wonder if we can have a "FilterReader function" that gets passed into this class instead and that gets called in readWithFilter, and when the vector reader ins enable it's the VectorFilterReader and a DefaultFilterReader. then we could pass any required fields as function arguments?